bolts/data/datasets/audio/ecoacoustics.py

```diff
@@ -1 +1,113 @@
+    """Dataset for audio data collected in various geographic locations."""
+
+    INDICES_DIR: ClassVar[str] = "AvianID_AcousticIndices"
+    METADATA_FILENAME: ClassVar[str] = "metadata.csv"
+
+    _FILE_INFO: ClassVar[FileInfo] = FileInfo(name="Ecoacoustics.zip", id="PLACEHOLDER")
+    _BASE_FOLDER: ClassVar[str] = "Ecoacoustics"
+    _EC_LABELS_FILENAME: ClassVar[str] = "EC_AI.csv"
+    _UK_LABELS_FILENAME: ClassVar[str] = "UK_AI.csv"
+
+    @parsable
+    def __init__(
+        self,
+        root: Union[str, Path],
+        *,
+        download: bool = True,
+        target_attr: SoundscapeAttr = "habitat",
+        transform: Optional[AudioTform] = None,
+    ) -> None:
+
+        self.root = Path(root)
+        self.download = download
+        self._base_dir = self.root / self._BASE_FOLDER
+        self.labels_dir = self._base_dir / self.INDICES_DIR
+        self._metadata_path = self._base_dir / self.METADATA_FILENAME
+        self.ec_labels_path = self.labels_dir / self._EC_LABELS_FILENAME
+        self.uk_labels_path = self.labels_dir / self._UK_LABELS_FILENAME
+
+        # Check data is downloaded and unzipped.
+        if self.download:
+            pass  # TODO: Implement download util function.
+        elif not self._files_unzipped():
+            raise RuntimeError(
+                f"Data not found at location {self._base_dir.resolve()}. " "Have you downloaded it?"
+            )
+
+        if not self._metadata_path.exists():
+            self._extract_metadata(target_attr)
+
+        self.metadata = pd.read_csv(self._base_dir / self.METADATA_FILENAME)
+
+        x = self.metadata["filePath"].to_numpy()
+        y = torch.as_tensor(self.metadata[f'{target_attr}_le'])
+        s = None
+
+        super().__init__(x=x, y=y, s=s, transform=transform, audio_dir=self._base_dir)
+
+    def _files_unzipped(self) -> bool:
+        dir_ = self.root / self._BASE_FOLDER
+        zip_checks = [
+            zipfile.is_zipfile(join(dir_, f)) for f in listdir(dir_) if isfile(join(dir_, f))
+        ]
+        return True not in zip_checks
+
+    def _label_encode_metadata(self, metadata: pd.DataFrame) -> pd.DataFrame:
+        """Label encode the extracted concept/context/superclass information."""
+        for col in metadata.columns:
+            # Skip over filepath and filename columns
+            if "file" not in col:
+                # Add a new column containing the label-encoded data
+                metadata[f"{col}_le"] = metadata[col].factorize()[0]
+        return metadata
+
+    def _extract_metadata(self, target_attr) -> None:
+        self.log("Extracting metadata.")
+        waveform_paths = list(self._base_dir.glob("**/*.wav"))
+
+        # Extract filepaths and names.
+        waveform_paths_str = [str(wvfrm.relative_to(self._base_dir)) for wvfrm in waveform_paths]
+        filepaths = pd.Series(waveform_paths_str)
+        pd.Series([path_str.split(os.sep)[-1] for path_str in waveform_paths_str])
```

Comment:
os not imported so os.sep won't work. If you used waveform_paths instead of waveform_path_str, then you could just do `pd.Series([pth.name for pth in waveform_paths])` using [Path.name](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name)

Reply:
Myles recommended I replace all that splitting stuff wit pandas string functions which I have done, no os.sep or os at all. might be in ecoacoustics-preprocessing commits but is definitely fixed.